pkgs/development/libraries/arrow-cpp/default.nix

```diff
@@ -156,6 +165,10 @@ stdenv.mkDerivation rec {
     "-DARROW_COMPUTE=ON"
     "-DARROW_CSV=ON"
     "-DARROW_DATASET=ON"
+    "-DARROW_ENGINE=${if enableEngine then "ON" else "OFF"}"
+    "-DARROW_FILESYSTEM=${if enableFilesystem then "ON" else "OFF"}"
+    "-DARROW_FLIGHT_SQL=${if enableFlightSql then "ON" else "OFF"}"
+    "-DARROW_IPC=${if enableIpc then "ON" else "OFF"}"
```

Comment:
If there is no more "business logic" for these options, should we just hardcode "ON" value? I'm generally suggesting reducing number of options as we can't test all of the combinations anyway.

Reply:
Yeah. I was thinking that maybe not everyone using this package will want the kitchen sink since it's a rather large sink, but I suppose they have the option overriding `cmakeFlagsArray`.

Reply:
Yes, the generic override interface already provides all the flexibility.

Reply:
Removed all of these flags.